### PR TITLE
fix: build system update

### LINKED ISSUE #97: Automatic upload of releases to PyPI via Travis
This would be really handy. It allows any one of us admins to tag a release and get it to PyPI with no extra work.

Can one of you maintainers do this? We have it for most of the other packages.

Thanks a lot.

### PR DESCRIPTION
**Pull request by henryiii**
Closes #78. Closes #97. Follows the Scikit-HEP developer guidelines.

TODO: activate Read-the-docs PR mode so we don't have to build that in CI.

### CODE REVIEW

**Comment by henryiii** [comment]
Who owns RtD for this? It needs to have build-on-PRs turned on.

**Comment by henryiii** [comment]
I'll check the wheel build after this gets merged.

**Comment by henryiii** [comment]
@HDembinski @piti118 @cdeil or @alexpearce, please add `scikit-hep` as a maintainer or owner for this package on PyPI. All Scikit-HEP packages should have `scikit-hep`. Thanks!

**Comment by henryiii** [comment]
Okay, ready.

File: setup.cfg
Comment (henryiii) [inline setup.cfg:12]:
```suggestion
    Development Status :: 7 - Inactive
```
?

Comment (henryiii) [reply setup.cfg:12]: We should probably add the standard Scikit-HEP maintainer lines here too.

**Comment by henryiii** [comment]
The badge needs to be updated to GHA

File: setup.cfg
Comment (eduardo-rodrigues) [reply setup.cfg]: Yep, makes very much sense.

**Comment by eduardo-rodrigues** [comment]
> @HDembinski @piti118 @cdeil or @alexpearce, please add `scikit-hep` as a maintainer or owner for this package on PyPI. All Scikit-HEP packages should have `scikit-hep` listed. Thanks!

+1: that would be great as many of us can then manage the package.

**Review by eduardo-rodrigues — APPROVED** [review]
This looks great and is such an improvement 👍 !

**Comment by eduardo-rodrigues** [comment]
> @HDembinski @piti118 @cdeil or @alexpearce, please add `scikit-hep` as a maintainer or owner for this package on PyPI. All Scikit-HEP packages should have `scikit-hep` listed. Thanks!

@henryiii I created a separate issue (https://github.com/scikit-hep/probfit/issues/106) for this, to move forward.

**Comment by eduardo-rodrigues** [comment]
> Who owns RtD for this? It needs to have build-on-PRs turned on.

I guess at least @HDembinski ?

**Comment by eduardo-rodrigues** [comment]
BTW this PR also closes https://github.com/scikit-hep/probfit/pull/103.

**Comment by HDembinski** [comment]
> > @HDembinski @piti118 @cdeil or @alexpearce, please add `scikit-hep` as a maintainer or owner for this package on PyPI. All Scikit-HEP packages should have `scikit-hep` listed. Thanks!
> 
> +1: that would be great as many of us can then manage the package.

I invited scikit-hep on PyPI two days ago when I saw the message. It says invitation pending.

**Comment by HDembinski** [comment]
Regarding RTD, I have the rights to add more maintainers. Which user should I invite?

**Comment by HDembinski** [comment]
I am perfectly happy to merge this, but I am not considering myself as the maintainer of this package. I haven't looked into it for a while. In other words, there is no need to wait for me to merge this, just merge as you see fit.

**Comment by eduardo-rodrigues** [comment]
> > > @HDembinski @piti118 @cdeil or @alexpearce, please add `scikit-hep` as a maintainer or owner for this package on PyPI. All Scikit-HEP packages should have `scikit-hep` listed. Thanks!
> > 
> > 
> > +1: that would be great as many of us can then manage the package.
> 
> I invited scikit-hep on PyPI two days ago when I saw the message. It says invitation pending.

Thanks very much @HDembinski. I got no notification and had not checked. Have just logged in and accepted the invitation. Sorted.

**Comment by eduardo-rodrigues** [comment]
> I am perfectly happy to merge this, but I am not considering myself as the maintainer of this package. I haven't looked into it for a while. In other words, there is no need to wait for me to merge this, just merge as you see fit.

Sounds more than fair. Indeed there is no maintainer per se anymore. On the other hand you do have/had certain admin rights, hence the messages to you.

**Comment by eduardo-rodrigues** [comment]
> Regarding RTD, I have the rights to add more maintainers. Which user should I invite?

Maybe add @henryiii since he is doing all the updates here?

**Comment by HDembinski** [comment]
Maybe we need some community rules for packages like this. If there are two scikit-hep people, the proponent of a change and another, approve the change, it can be merged by the proponent?

**Comment by HDembinski** [comment]
> > Regarding RTD, I have the rights to add more maintainers. Which user should I invite?
> 
> Maybe add @henryiii since he is doing all the updates here?

Done

**Comment by eduardo-rodrigues** [comment]
> Maybe we need some community rules for packages like this. If there are two scikit-hep people, the proponent of a change and another, approve the change, it can be merged by the proponent?

I pretty much do like that for work with Henry on Particle, DecayLanguage, etc. In fact, if I review and it is a no-brainer I do the merge myself straight away most of the time; else if there is just a little question or update I approve, Henry makes the update to his PR and merged once the CI is again green. In short, I agree with your pragmatic rule and think we mostly follow it implicitly everywhere where more than 2 person is a main author and/or maintainer.

**Comment by HDembinski** [comment]
Closes #97

**Comment by henryiii** [comment]
Thanks everyone!

> Maybe we need some community rules for packages like this. If there are two scikit-hep people, the proponent of a change and another, approve the change, it can be merged by the proponent?

That's the unofficial way that we tend to work on "community maintained" packages. Might be better to make it a bit more formal and explicit; though so far, it's mostly @eduardo-rodrigues and myself doing community maintenance. For packages with one or more dedicated maintainers (iMinuit, boost-histogram, uproot, mplhep, etc), then lead maintainer gets final say (except in an emergency like a broken release and they can't be reached? But so far all our maintainers have been good about avoiding releasing broken packages and have been very accessible).

**Comment by eduardo-rodrigues** [comment]
Makes sense, @henryiii.

**Pull request merged**
